logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/ClientRepository.kt

```diff
@@ -134,14 +141,29 @@ class ClientDataSource(
         return clientRemoteRepository.deleteClient(param)
     }
 
-    // TODO(self-device-list): after fetch save list of self client in the db
+    /**
+     * fetches the clients from the backend and stores them in the database
+     */
     override suspend fun selfListOfClients(): Either<NetworkFailure, List<Client>> {
-        return clientRemoteRepository.fetchSelfUserClients()
+        return wrapApiRequest { clientApi.fetchSelfUserClient() }
+            .onSuccess { clientList ->
+                val selfUserIdDTO = selfUserID.toApi()
+                val list = clientList.map { clientMapper.toInsertClientParam(it, selfUserIdDTO) }
+                // when calling this function the first time after tooManyClients error
```

Comment:
Why does this fail? Because the self user doesn't yet exist in the DB and you can't insert a client without referencing a user?

Reply:
Exactly, we can either ignore it or do it this way where we try insert the user ID first 
https://github.com/wireapp/kalium/pull/1553/files#diff-67fa9eaf81a27a0ce3268570db69f85d76d9e1f5707010df43cb1e1e92c8a89aR70

Reply:
Seems wrong to insert the user in the ClientDAO though? Why not include `UserDAO` here and insert the user here at the repository level. Also don't we have the same issue for `clientDAO.insertClients()` method?

Reply:
i reverted this change

Reply:
> i reverted this change

You forgot to also revert the sql query